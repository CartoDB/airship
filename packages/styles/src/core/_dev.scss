@import 'tools/_mixins';
@import 'variables/_variables';
