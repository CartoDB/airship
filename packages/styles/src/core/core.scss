@import 'variables/_variables';

* {
  box-sizing: border-box;
  font-family: sans-serif;
}

:root {
  --primary-color: #{$primary-color};
}
