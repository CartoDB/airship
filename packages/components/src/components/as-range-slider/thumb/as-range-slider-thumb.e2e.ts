
import { E2EPage, newE2EPage } from '@stencil/core/testing';

describe('as-range-slider-thumb', async () => {
  let page: E2EPage;

  beforeEach(async () => {
    page = await newE2EPage();

    await page.setContent(`
      <as-range-slider-thumb
        percentage="50"
        value="50"
        valueMin="0"
        valueMax="100"></as-range-slider-thumb>
      <div class="as-range-slider__rail"></div>
    `);
  });

  describe('Rendering', async () => {
      it('should render a text with its value', async () => {
        const element = await page.find('as-range-slider-thumb');
        element.setProperty('value', 5);
        element.setProperty('formatValue', () => `${element.getProperty('value')}â‚¬`);
        await page.waitForChanges();

        expect(element.outerHTML).toMatchSnapshot();
      });

      it('should have aria attributes', async () => {
        const element = await page.find('as-range-slider-thumb');
        element.setProperty('value', 5);
        element.setProperty('valueMin', 0);
        element.setProperty('valueMax', 10);
        element.setProperty('percentage', 10);
        await page.waitForChanges();

        expect(element.outerHTML).toMatchSnapshot();
      });
  });

  describe('Interaction', async () => {
    it('should emit thumbIncrease / thumbDecrease events when using keyboard arrows', async () => {
      await page.waitForChanges();
      const onThumbIncreaseSpy = await page.spyOnEvent('thumbIncrease');
      const onThumbDecreaseSpy = await page.spyOnEvent('thumbDecrease');

      const element = await page.find('.as-range-slider__thumb');

      await element.press('ArrowLeft');
      await element.press('ArrowRight');

      expect(onThumbDecreaseSpy).toHaveReceivedEvent();
      expect(onThumbIncreaseSpy).toHaveReceivedEvent();
    });

    it('should emit changeStart event when using mouse', async () => {
      const thumbChangeStart = await page.spyOnEvent('thumbChangeStart');

      const element = await page.find('.as-range-slider__thumb-handle');
      await element.click();

      expect(thumbChangeStart).toHaveReceivedEvent();
    });
  });
});
