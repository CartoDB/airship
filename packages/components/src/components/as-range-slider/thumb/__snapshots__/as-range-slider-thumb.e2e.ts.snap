// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`as-range-slider-thumb Rendering should have aria attributes 1`] = `"<as-range-slider-thumb percentage=\\"50\\" value=\\"50\\" valuemin=\\"0\\" valuemax=\\"100\\" class=\\"hydrated\\"><div role=\\"slider\\" tabindex=\\"0\\" aria-valuetext=\\"5\\" aria-valuenow=\\"5\\" class=\\"as-range-slider__thumb\\" data-value=\\"5\\" aria-valuemin=\\"0\\" aria-valuemax=\\"10\\" style=\\"left: 10%;\\"><div class=\\"as-range-slider__thumb-handle\\"></div><span class=\\"as-caption as-font-medium as-range-slider__value\\">5</span></div></as-range-slider-thumb>"`;

exports[`as-range-slider-thumb Rendering should render a text with its value 1`] = `"<as-range-slider-thumb percentage=\\"50\\" value=\\"50\\" valuemin=\\"0\\" valuemax=\\"100\\" class=\\"hydrated\\"><div role=\\"slider\\" tabindex=\\"0\\" aria-valuetext=\\"5\\" aria-valuenow=\\"5\\" class=\\"as-range-slider__thumb\\" data-value=\\"5\\" style=\\"left: 50%;\\"><div class=\\"as-range-slider__thumb-handle\\"></div><span class=\\"as-caption as-font-medium as-range-slider__value\\">5</span></div></as-range-slider-thumb>"`;
